src/Compiler/LatteToPhpCompiler.php

```diff
@@ -94,6 +93,69 @@ public function compileFile(?string $actualClass, string $templatePath, array $v
         return $compileFilePath;
     }
 
+    /**
+     * @param Variable[] $variables
+     * @param Component[] $components
+     * @param CollectedForm[] $forms
+     */
+    private function compile(?string $actualClass, string $templateContent, array $variables, array $components, array $forms, string $context = ''): string
+    {
+        $phpContent = $this->compiler->compile($templateContent, $actualClass, $context);
+        $phpContent = $this->explicitCalls($actualClass, $phpContent, $variables, $components);
+        $phpContent = $this->addExtractParams($phpContent);
+        $phpContent = $this->addFormClasses($phpContent, $forms);
+        return $this->remapLines($phpContent);
+    }
+
+    /**
+     * @param CollectedForm[] $forms
+     */
+    private function addFormClasses(string $phpContent, array $forms): string
+    {
+        $componentType = 'Nette\ComponentModel\IComponent';
+        $addedForms = [];
+        foreach ($forms as $form) {
+            $formName = $form->getName();
+            // TODO check why there are 5 forms instead of one
+            if (isset($addedForms[$formName])) {
+                continue;
+            }
+            $addedForms[$form->getName()] = true;
+
+            $method = (new Method('offsetGet'))
+                ->addParam(new Param('name'))
+                ->addStmts([
+                    new Return_(
+                        new StaticCall(
+                            new Name('parent'),
+                            new Identifier('offsetGet'),
+                            [
+                                new Arg(new NodeVariable('name')),
+                            ]
+                        )
+                    ),
+                ])
+                ->makePublic()
+                ->setReturnType($componentType);
```

Comment:
This is really ugly :-D I would try to came up with better solution. But it is good enough.

Reply:
maybe we can use array shapes for this? Or make it as properties and replace array access with property access? `...->form["field"]` to `...->form->field`

Reply:
Yes it is, this whole PR is just POC if we can handle forms :) there are lot of str / preg replaces etc.

Reply:
Yes we probably can change all `end($formStack)["something"]` to $form->something and change conditional return type to `@property-read` comment in fake Form class

Reply:
I would preffer it that way in final version